RealestateApi/Service/AgentService.cs

```diff
@@ -36,5 +36,23 @@ public async Task<ServiceResult<AgentModel>> GetAgentByIdAsync(int agentId)
         {
             return await _agentRepository.GetAgentByIdAsync(agentId);
         }
+
+        public async Task<ServiceResult<AgentModel>> DeleteAgentByIdAsync(int agentId)
+        {
+            ServiceResult<AgentModel> response = new();
+
+            var existAgent = await GetAgentByIdAsync(agentId);
```

Comment:
You check if the agent exists two times.
One here and one inside agent repository (when you call the delete).
Remove this line because it's already being executed in deleteAgentById from agent repository

Reply:
Done